operate/pom.xml

```diff
@@ -257,6 +256,19 @@
         </plugins>
       </build>
     </profile>
-  </profiles>
 
+    <!-- 
+      Integration tests are run per default, but having it in a profile allows us to exclude it easily.
+      That is useful for example for creating releases.
+       -->
+    <profile>
+      <id>integration-tests</id>
+      <activation>
+        <activeByDefault>true</activeByDefault>
```

Comment:
💭  we already have the `skipTests` property can't we use that for ITs too? then you wouldn't need that profile and have one property to disable all sorts of tests?

Reply:
No because this only skips tests but still compiles and generates gpg and all these sort of stuff.

Reply:
Yeah, we really want to completely exclude the whole module when releasing, so profiles usually are the best way to go. Or skipping the plugins in those modules that you want to avoid running (e.g. gpg signing, deploying, etc.). Depends a bit on much you want to skip. Usually, it's clearer and cleaner using profiles like this.

Reply:
> 💭 I think it's fine here generally. However, as soon as you enable ANY other profile, this won't trigger anymore. That is a typical "what the heck is going on?" moment with Maven 😄

Expanding on this, this even happens when specific os- or JDK version-specific profiles are activated. Those `activeByDefault` profiles don't run by default anymore then. 😄

Reply:
We could use a rather hacky activation that always runs the QA as well unless you explicitly exclude it:
```suggestion
        <file>
          <exists>pom.xml</exists>
        </file>
```

Reply:
Is `-Dmaven.test.skip=true` an option? that should skip compilation altogether.

Reply:
That is interesting this would at least also avoid for the other modules. 👍

 But my point for the profile is to really do nothing with these modules as you would still verify and generate sources for these, right?

But I'm also fine to remove the profile as due to the other refactoring I did later it is maybe not that much of an impact anymore (just unnecessary work we do)

Reply:
@oleschoenburg I tried this shortly via `-Dmaven.test.skip=true` https://github.com/camunda/zeebe/pull/17497/commits/0b9a1c3eef3f12d73f16e2aeb493999f80244253

I ran into the [same issue](https://github.com/camunda/zeebe/actions/runs/8774388988/job/24075524714#step:14:10597) when I tried to compile only the project during prepare (I tested several approaches earlier). The issue is that the [snapshot project depends on a testjar that is not compiled if we use this flag](https://github.com/camunda/zeebe/blob/main/zeebe/snapshot/pom.xml#L65-L71) (or use only compile), that is why I had to do this with the profile. 

I see it as the only option right now, and would like to go ahead with it. If you see a different option please feel free to improve it afterwards.